controllers/nnf_access_controller.go

```diff
@@ -170,65 +170,113 @@ func (r *NnfAccessReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	if access.Status.State == "mounted" {
+		result, err := r.mount(ctx, access, clientList, storageMapping)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if result != nil {
+			return *result, nil
+		}
+	} else {
+		result, err := r.unmount(ctx, access, clientList, storageMapping)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if result != nil {
+			return *result, nil
+		}
```

Comment:
Could you move these if tests outside the block and handle it once for both cases?